tests/foreign_object/test_tuple_lookups.py

```diff
@@ -0,0 +1,242 @@
+        cls.customer_2 = Customer.objects.create(customer_id=1, company="b")
+        cls.customer_3 = Customer.objects.create(customer_id=2, company="c")
+        cls.customer_4 = Customer.objects.create(customer_id=3, company="d")
+        cls.customer_5 = Customer.objects.create(customer_id=1, company="e")
+        cls.contact_1 = Contact.objects.create(customer=cls.customer_1)
+        cls.contact_2 = Contact.objects.create(customer=cls.customer_1)
+        cls.contact_3 = Contact.objects.create(customer=cls.customer_2)
+        cls.contact_4 = Contact.objects.create(customer=cls.customer_3)
+        cls.contact_5 = Contact.objects.create(customer=cls.customer_1)
+        cls.contact_6 = Contact.objects.create(customer=cls.customer_5)
+
+    def test_exact(self):
+        test_cases = (
+            (self.customer_1, (self.contact_1, self.contact_2, self.contact_5)),
+            (self.customer_2, (self.contact_3,)),
+            (self.customer_3, (self.contact_4,)),
+            (self.customer_4, ()),
+            (self.customer_5, (self.contact_6,)),
+        )
+
+        for customer, contacts in test_cases:
+            with self.subTest(customer=customer, contacts=contacts):
+                self.assertSequenceEqual(
+                    Contact.objects.filter(customer=customer).order_by("id"), contacts
+                )
+
+    def test_exact_subquery(self):
+        with self.assertRaisesMessage(
+            NotSupportedError, "'exact' doesn't support multi-column subqueries."
+        ):
+            subquery = Customer.objects.filter(id=self.customer_1.id)[:1]
+            self.assertSequenceEqual(
+                Contact.objects.filter(customer=subquery).order_by("id"), ()
+            )
+
+    def test_in(self):
+        cust_1, cust_2, cust_3, cust_4, cust_5 = (
+            self.customer_1,
+            self.customer_2,
+            self.customer_3,
+            self.customer_4,
+            self.customer_5,
+        )
+        c1, c2, c3, c4, c5, c6 = (
+            self.contact_1,
+            self.contact_2,
+            self.contact_3,
+            self.contact_4,
+            self.contact_5,
+            self.contact_6,
+        )
+        test_cases = (
+            ((), ()),
+            ((cust_1,), (c1, c2, c5)),
+            ((cust_1, cust_2), (c1, c2, c3, c5)),
+            ((cust_1, cust_2, cust_3), (c1, c2, c3, c4, c5)),
+            ((cust_1, cust_2, cust_3, cust_4), (c1, c2, c3, c4, c5)),
+            ((cust_1, cust_2, cust_3, cust_4, cust_5), (c1, c2, c3, c4, c5, c6)),
+        )
+
+        for contacts, customers in test_cases:
+            with self.subTest(contacts=contacts, customers=customers):
+                self.assertSequenceEqual(
+                    Contact.objects.filter(customer__in=contacts).order_by("id"),
+                    customers,
+                )
+
+    @unittest.skipIf(
+        connection.vendor == "mysql",
+        "MySQL doesn't support LIMIT & IN/ALL/ANY/SOME subquery",
+    )
```

Comment:
Instead of `connection.vendor` checks, please use [DatabaseFeatures.django_test_skips](https://github.com/django/django/blob/2c029c718f45341cdd43ee094c24488743c633e6/django/db/backends/mysql/features.py#L94) instead.

Reply:
We actually have a feature switch for that folks https://github.com/django/django/blob/8864125d1f423ee94c2bb1cc36ea998619d47c3f/django/db/backends/mysql/features.py#L10-L11

Reply:
Should be handled by https://github.com/django/django/pull/18745